### PR TITLE
Need to join channels before setting their topic

### PR DESCRIPTION
**Pull request by royrapoport**
we need to automatically join channels we'll set topics for.  So

1. Added join_channel() method to slacker.py
2. Added leave_channel() method to slacker.py
3. set_topic() now automatically calls join_channel()
4. added a `leave` optional parameteer to set_topic() and conditional_set_topic(); if set, we will automatically leave the channel after setting the topic.
5. Modified call of conditional_set_topic() in user_and_channel_report to specify `leave` == True

### CODE REVIEW

**Comment by jdq22** [comment]
How did this come up? You're already in all these channels because you need to be able to post to them, no?

**Comment by royrapoport** [comment]
Ha! No, because slackbot posts in them :)

File: user_and_channel_report [hunk 1]
```diff
@@ -49,8 +49,7 @@ for channel in channels:
         dest = channel_dict[channel]
         name = cobj.get(channel)['name']
         nt = topic.format(name)
-        sobj.conditional_set_topic(dest, nt)
+        sobj.conditional_set_topic(dest, nt, leave=True)
```
Comment (jdq22) [inline user_and_channel_report:52]: Do you want to stay in the channels you're already in? I think this is going to make you leave every -stats channel.

Comment (royrapoport) [reply user_and_channel_report:52]: Yeah, that's correct.  I don't much care about being in the -stats channels.

**Review by jdq22 — APPROVED** [review]

**Pull request merged**